lib/plugins/aws/package/compile/events/httpApi.js

```diff
@@ -41,6 +41,18 @@ class HttpApiEvents {
     serverless.httpApiEventsPlugin = this;
 
     this.hooks = {
+      'initialize': () => {
+        if (
+          this.serverless.service.provider.name === 'aws' &&
```

Comment:
I believe we should go a bit different route here. If someone uses `provider.httpApi.useProviderTags` setting, we shouldn't show deprecation, because with next major nothing will change for these users. Instead, we should emit a deprecation to people that will be affected, so the ones that have at least one function with an `httpApi` event, are using `provider.tags` but don't have `provider.httpApi.useProviderTags` flag set - these users when migrating to next major, will experience a change and the tags will be automatically applied to their API Gateway, potentially removing any other tags set in a different way.

Reply:
modified it accordingly and thanks for the detailed explanation